docs/coding_guidelines.md

```diff
@@ -0,0 +1,80 @@
+# Coding guidelines for OCS-CI project
+
+Lets follow the following guidelines for the Core Libraries and Test Scripts
+
+* Follow pep8 or use autopep8 to automagically conform to pep8 style.
+* Each Python Class is well documented for inheritence, methods and
+    attributes, Add sufficent comments for code block that is easy
+    to understand for others to review.
+    Use [Google Style of Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
+* Code Readablity takes priority, Lets avoid crazy hacks and tricks 
+    and stick to straight forward code blocks that are easy to
+    understand for all.
+* For core libraries when possible please add the unittest to
+    minimize regression for many tests
+* Helper functions that are useful to wide variety of tests should
+    reside in utils and/or should be carefully reviewed to check if
+    they can actually be methods.
+* test_scripts will reside in test/ folders and will define run 
+    unction
+* test_scripts should return non zero exit status for failure and 0
+    for 'pass'
+
+**ex: test_scripts/test_ocs_operator.py**
+
+```python
+        def run(ocs_context, **kw):
+            # use ocs_context to deal with cluster
+            # run your test steps
+            if test_failed:
+                return False
+            else:
+                return Pass
+```
+
+* Do not use backslashes in theh code for line breaker!
+* Line length should be maximum 79 characters!
+* Try to follow this convention for brackets and indentions like in example below (This makes code much more readable):
```

Comment:
I think we want this when it cannot fit in one line.

Reply:
probably we can add both examples to make it clear, one which fits in single line and one which doesn't

Reply:
Added another example and mentioned about if it doesn't fit 79 chars.